libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPlugin.kt

```diff
@@ -248,6 +250,18 @@ abstract class AbstractKotlinPlugin @Inject constructor(val scriptHandler: Scrip
             }
         })
     }
+
+    private fun configureResolutionStrategy(project: Project) {
+        val projectVersion = loadKotlinVersionFromResource(project.logger)
+        project.configurations.all(Action<Configuration> { configuration ->
+            configuration.resolutionStrategy.eachDependency { details ->
+                val requested = details.requested
+                if (requested.group.equals("org.jetbrains.kotlin") && requested.version.isEmpty()) {
+                    details.useTarget("${requested.group}:${requested.name}:$projectVersion")
```

Comment:
here it is not clear which project version it will be: kotlin's version or user's project version. So I think you better to rename it and ensure it actually contains valid version (you can change sample project's version to some particular value (for example, 1.0.0))